tests/FSharp.Test.Utilities/Compiler.fs

```diff
@@ -1425,6 +1425,18 @@ Actual:
                     failwith $"""Output does not match expected:{Environment.NewLine}{item}{Environment.NewLine}Actual:{Environment.NewLine}{actual}{Environment.NewLine}"""
             cResult
```

Comment:
You only really used `VerifyNotInOutput` in the test.
The other added functions are not used and can be removed.

Also, add release notes for this new flag and new feature.

Also, do run tests via `./build.sh -c Release --testcoreclr --norestore` and address all failures.